src/gm-apclientpp.cpp

```diff
@@ -29,15 +29,29 @@
 #define GM_FALSE (0.)
 #define GM_TRUE (1.)
 #define GM_BOOL(cond) ((cond) ? GM_TRUE : GM_FALSE)
+#define JSON_MISSING -1.
+#define JSON_OBJECT 0.
+#define JSON_ARRAY 1.
+#define JSON_STRING 2.
+#define JSON_NUMBER 3.
+#define JSON_NULL 4.
 
 using json = nlohmann::json;
 
+typedef struct {
+    std::string script;
+    std::string script_name;
+    json script_data;
+} PollEvent;
+
 
 static int api = 0;
 static std::unique_ptr<APClient> apclient;
 static std::string result; // buffer for string results of simple functions
-static std::queue<std::string> queue; // queue of events to run on poll
+static std::queue<PollEvent> queue; // queue of events to run on poll
 static std::string script; // buffer for script result
+static std::string script_name; // buffer for the script's name
+static std::vector<json> script_data;  // buffer for script data
```

Comment:
One thing to consider from a performance standpoint is that writing here will make a copy of the json, rather than just a reference. I doubt this will be a huge bottle-neck in performance, but if you see slowdown, I'd recommend changing that to `const json*` and put the root json node into a separate variable (that's not a pointer).